1. For the suburb of Altona, it is postulated that a typical property sells for $800,000.
• Use the data at hand to test this assumption.
• Is the typical property price really 800000 or has it increased?
• Use a significance level of 5%.

In [5]:
import pandas as pd
import math
from scipy import stats

# Load dataset
df = pd.read_csv("/Users/dholakiyan/De/property.csv")

# Filter data for Altona suburb
altona = df[df["Suburb"] == "Altona"]

# Get property prices
prices = altona["Price"].dropna()

# Sample statistics
n = len(prices)
mean_price = prices.mean()
std_dev = prices.std(ddof=1)

# Hypothesized mean
mu = 800000

# t-test statistic
t_stat = (mean_price - mu) / (std_dev / math.sqrt(n))

# One-tailed p-value
p_value = 1 - stats.t.cdf(t_stat, df=n-1)

print("Sample size:", n)
print("Mean price:", mean_price)
print("Standard deviation:", std_dev)
print("t-statistic:", t_stat)
print("p-value:", p_value)

Sample size: 74
Mean price: 834830.4054054054
Standard deviation: 291546.0454736785
t-statistic: 1.0277020770199676
p-value: 0.15374163565277732


Hypothesis Test

H₀: Typical property price = $800,000
H₁: Typical property price > 800,000
α = 0.05
p-value = 0.1537 > 0.05
We fail to reject the null hypothesis.

At the 5% significance level, there is not enough statistical evidence to conclude that the typical property price in Altona is greater than $800,000.

The sample mean is $834,830, but the increase is not statistically significant.

2. For the year 2016, is there any difference in the prices of properties sold in the summer months vs winter months?
• Consider months from October till March as winter months and rest as summer months.
• Use a significance level of 5%.

Hypothesis

H₀ (Null Hypothesis): There is no difference in mean property prices between summer and winter.
μ(summer)=μ(winter)

H₁ (Alternative Hypothesis): There is a difference in mean prices.
μ(summer)=!μ(winter)

Significance level:
α=0.05

In [8]:
import pandas as pd
from scipy.stats import ttest_ind
import warnings
warnings.filterwarnings('ignore')

df = pd.read_csv("/Users/dholakiyan/Desktop/property.csv")

# Fix date parsing
df["Date"] = pd.to_datetime(df["Date"], errors="coerce")

df = df.dropna(subset=["Date"])

# Filter year
df_2016 = df[df["Date"].dt.year == 2016]

# Extract month
df_2016["Month"] = df_2016["Date"].dt.month

# Winter (Oct–Mar)
winter = df_2016[df_2016["Month"].isin([10,11,12,1,2,3])]["Price"]

# Summer (Apr–Sep)
summer = df_2016[~df_2016["Month"].isin([10,11,12,1,2,3])]["Price"]

# t-test
t_stat, p_value = ttest_ind(summer, winter, equal_var=False)

print("Summer mean:", summer.mean())
print("Winter mean:", winter.mean())
print("p-value:", p_value)

Summer mean: 1020054.3357933579
Winter mean: 1144211.7860230547
p-value: 2.278917771002362e-06


Using a two-sample t-test at a 5% significance level, the p-value obtained is 2.28×10⁻⁶, which is less than 0.05. Therefore, we reject the null hypothesis and conclude that there is a significant difference in property prices between summer and winter months in 2016, with winter prices being higher.

3. For the suburb of Abbotsford, what is the probability that out of 10 properties sold, 3 will not have a car parking space?
• Use the column car in the dataset.
• Round off your answer to 3 decimal places.

In [13]:
import pandas as pd
from scipy.stats import binom

# Load dataset
df = pd.read_csv("/Users/dholakiyan/Desktop/property.csv")

# Filter suburb Abbotsford
abbotsford = df[df["Suburb"] == "Abbotsford"]

# Probability that a property has NO car parking
p = (abbotsford["Car"] == 0).mean()

# Binomial probability: exactly 3 out of 10
n = 10
k = 3

probability = binom.pmf(k, n, p)

print("Probability of no parking:", p)
print("Final probability:", round(probability, 3))

Probability of no parking: 0.26785714285714285
Final probability: 0.26


Based on the dataset, the probability that a property in Abbotsford has no car parking space is 0.260. Using the binomial distribution for 10 properties, the probability that exactly 3 properties will not have a car parking space is approximately 0.26 (26%) when rounded to three decimal places.

4. In the suburb of Abbotsford, what are the chances of finding a property with 3 rooms? Round your answer to 3 decimal places.

In [14]:
import pandas as pd

# Load dataset
df = pd.read_csv("/Users/dholakiyan/Desktop/property.csv")

# Filter Abbotsford properties
abbotsford = df[df["Suburb"] == "Abbotsford"]

# Total properties in Abbotsford
total_properties = len(abbotsford)

# Properties with 3 rooms
three_rooms = len(abbotsford[abbotsford["Rooms"] == 3])

# Probability
probability = three_rooms / total_properties

print("Probability:", round(probability, 3))

Probability: 0.357


Conclusion:
The probability of finding a property with 3 rooms in Abbotsford is approximately 0.357, meaning about 35.7% of properties in Abbotsford have 3 rooms based on the dataset.

5. In the suburb of Abbotsford, what are the chances of finding a property with 2 bathrooms? Round your answer to 3 decimal places.

In [6]:
import pandas as pd

# Load dataset
df = pd.read_csv("property.csv")

# Filter Abbotsford properties
abbotsford = df[df["Suburb"] == "Abbotsford"]

# Total properties
total = len(abbotsford)

# Properties with 2 bathrooms
two_bath = len(abbotsford[abbotsford["Bathroom"] == 2])

# Probability
probability = two_bath / total

print("Probability:", round(probability, 3))

Probability: 0.339


Conclusion: The probability of finding a property with 2 bathrooms in Abbotsford is approximately 0.339, meaning about 33.9% of properties in Abbotsford have two bathrooms, based on the dataset.

6. One-Sample Hypothesis Test (Industry Pricing) A real estate firm claims that the average property price in Richmond is $1,000,000. Using the dataset, test whether the actual average price is significantly different from this claim at a 5% significance level. Clearly state:
• Null and alternative hypotheses
• Test statistic
• p-value
• Final business conclusion

Null Hypothesis (H₀):
μ=1,000,000
The average property price in Richmond is $1,000,000.

Alternative Hypothesis (H₁):
𝜇≠1,000,000
The average property price in Richmond is different from $1,000,000.

Significance level:
α=0.05

In [16]:
import pandas as pd
from scipy import stats

# Load dataset
df = pd.read_csv("/Users/dholakiyan/Desktop/property.csv")

# Filter Richmond properties
richmond = df[df["Suburb"] == "Richmond"]["Price"]

# Hypothesized mean
mu = 1000000

# One-sample t-test
t_stat, p_value = stats.ttest_1samp(richmond, mu)

print("Sample mean:", richmond.mean())
print("Test statistic:", t_stat)
print("p-value:", p_value)

Sample mean: 1083564.423076923
Test statistic: 2.579547704074923
p-value: 0.010444990664152026


Conclusion:
At the 5% significance level, the p-value is less than 0.05, indicating strong evidence that the actual average property price in Richmond is significantly different from $1,000,000.

The dataset suggests that the average property price in Richmond is higher than the claimed value of $1,000,000.

7. Independent Two-Sample T-Test (Feature Impact)
Do properties with car parking sell at a higher average price than properties without car parking, across the entire dataset? Use a 5% significance level and justify:
• Choice of test
• Interpretation of p-value
• Business implications for developers

Choice of Test

An Independent Two-Sample t-test is used because:
We are comparing the mean prices of two independent groups:
1. Properties with car parking
2. Properties without car parking
The variable Price is continuous
The groups are independent observations

Hypothesis:
Null Hypothesis (H₀):

μ(parking)=μ(no parking)
The average price of properties with car parking is equal to the average price of properties without car parking.

Alternative Hypothesis (H₁):
μ(parking)>μ(no parking)
Properties with car parking sell at a higher average price.

Significance level:
α=0.05

In [17]:
import pandas as pd
from scipy.stats import ttest_ind

# Load dataset
df = pd.read_csv("/Users/dholakiyan/Desktop/property.csv")

# Separate groups
with_parking = df[df["Car"] > 0]["Price"]
without_parking = df[df["Car"] == 0]["Price"]

# Independent t-test
t_stat, p_value = ttest_ind(with_parking, without_parking, equal_var=False)

print("Mean price with parking:", with_parking.mean())
print("Mean price without parking:", without_parking.mean())
print("t-statistic:", t_stat)
print("p-value:", p_value)

Mean price with parking: 1074443.9241114312
Mean price without parking: 1079088.0107212476
t-statistic: -0.27224703025972535
p-value: 0.78547499853986


Interpretation of p-value
Significance level:
α=0.05

p-value:
p=0.785

Comparison:
0.785>0.05

Decision:
Fail to reject the null hypothesis (H₀).
This means there is no statistically significant difference in average property prices between properties with car parking and those without car parking in the dataset.
The negative t-statistic also indicates that the mean price without parking is slightly higher, but the difference is not statistically significant.

Business Implications for Developers:
Since the statistical test shows no significant price difference, the presence of car parking does not appear to strongly influence property prices in this dataset.

Implications:
Developers should not assume parking automatically increases property value.
Other factors such as location, number of rooms, property size, and neighborhood amenities may have a greater impact on price.
In some areas, buyers may prioritize affordability or proximity to public transport rather than parking availability.
Developers should analyze multiple property features together when determining pricing strategies.

8. Two-Way ANOVA (Location & Property Type)
Investigate whether property prices are influenced by:
• Suburb
• Type of property
• Interaction between suburb and property type
Use Two-Way ANOVA and explain which factors significantly affect price.

In [20]:
pip install statsmodels

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.0/10.0 MB 61.8 kB/s  0:02:1463.1 kB/s eta 0:00:05
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2/2 [statsmodels] 1/2 [statsmodels]
Note: you may need to restart the kernel to use updated packages.


In [4]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
import warnings
warnings.filterwarnings('ignore')

# Load dataset
df = pd.read_csv("/Users/dholakiyan/Desktop/property.csv")

# Two-way ANOVA model
model = ols('Price ~ C(Suburb) + C(Type) + C(Suburb):C(Type)', data=df).fit()

anova_table = sm.stats.anova_lm(model, typ=2)

print(anova_table)

                         sum_sq       df          F    PR(>F)
C(Suburb)          3.751850e+15    313.0  74.652789  0.000000
C(Type)                     NaN      2.0        NaN       NaN
C(Suburb):C(Type)  6.498312e+14    626.0   6.465038  0.011013
Residual           2.069703e+15  12890.0        NaN       NaN


Hypotheses

For each factor:
Suburb
H₀: Property price does not differ by suburb
H₁: Property price differs by suburb

Property Type
H₀: Property price does not differ by property type
H₁: Property price differs by property type

Interaction (Suburb × Type)
H₀: No interaction between suburb and property type
H₁: There is interaction between suburb and property type

Significance level:
α=0.05

Decision Rule:

If p-value < 0.05 → Significant effect
If p-value ≥ 0.05 → Not significant

Suburb Effect:
p-value for Suburb: 0.000000<0.05

Decision: Reject the null hypothesis.

Interpretation:
Property prices significantly differ across suburbs.
This means location plays an important role in determining property price.


Property Type Effect:
C(Type) → NaN
This usually happens when the dataset does not contain enough variation or has missing categories after filtering.

Interpretation:
The effect of property type cannot be reliably evaluated in this ANOVA output.
However, based on the current result, no statistical conclusion can be drawn for property type alone.


Interaction Effect (Suburb × Type)
p-value: 0.011013<0.05

Decision: Reject the null hypothesis.

Interpretation:
There is a significant interaction between suburb and property type.
This means the impact of property type on price depends on the suburb.

Conclusion:
The Two-Way ANOVA results show that suburb significantly influences property prices, indicating that location is a key determinant of property value.

Additionally, the interaction between suburb and property type is statistically significant, meaning the effect of property type on price varies across different suburbs.

For developers and investors, this suggests that pricing and development strategies should consider both the suburb and the type of property together. Certain property types may perform better in specific suburbs, and market demand can vary depending on location.

Therefore, location remains the strongest driver of property price, and the value of different property types depends on the suburb in which they are located.

9. p-Value Interpretation (Decision Making)
A hypothesis test comparing prices across two suburbs results in a p-value of 0.032.
Answer:
• What does this p-value indicate?
• Should the null hypothesis be rejected at α = 0.05?
• How should a business stakeholder interpret this result?

1. What does this p-value indicate?

The p-value (0.032) represents the probability of observing the data, or something more extreme, assuming the null hypothesis is true.

In this case, it means there is a 3.2% chance of observing the difference in property prices between the two suburbs if there were actually no real difference in their average prices.

Since this probability is relatively small, it suggests that the observed difference is unlikely to have occurred by random chance alone.

2. Should the null hypothesis be rejected at α=0.05?

Significance level:
α=0.05

Comparison:
p=0.032<0.05

Decision:
Reject the null hypothesis.

This indicates that there is statistically significant evidence that the average property prices between the two suburbs are different.

3. How should a business stakeholder interpret this result?

For a business stakeholder (such as a real estate developer or investor), this result suggests that location significantly affects property prices.

Implications include:
Property values vary meaningfully between suburbs.
Investors may achieve different returns depending on the suburb they choose.
Developers should consider location-based pricing strategies when planning projects.
Marketing and investment decisions should account for suburb-level market differences.
Overall, the result indicates that suburb selection is an important factor in property pricing and investment decisions.

10. Industry-Style Hypothesis Validation (Policy Decision)
A housing policy group believes that properties with more than 2 bathrooms command a premium price.
Design and execute a statistical test to validate this claim:
• Identify the correct test
• State hypotheses
• Report p-value
• Give a clear recommendation to policymakers

1. Identify the Correct Test

The appropriate statistical test is an Independent Two-Sample t-test.

Reason:
We are comparing the average prices of two independent groups:
1. Properties with more than 2 bathrooms
2. Properties with 2 or fewer bathrooms
The variable Price is continuous.
The groups are independent observations.

Significance level:
α=0.05

2. Hypotheses

Null Hypothesis (H₀):
μ>2(bath)≤μ≤2(bath)
Properties with more than 2 bathrooms do not have a higher average price.

Alternative Hypothesis (H₁):
μ>2(bath)>μ≤2(bath)
Properties with more than 2 bathrooms have a higher average price (premium).

In [5]:
import pandas as pd
from scipy.stats import ttest_ind

df = pd.read_csv("/Users/dholakiyan/Desktop/property.csv")

# Create groups
more_than_two = df[df["Bathroom"] > 2]["Price"]
two_or_less = df[df["Bathroom"] <= 2]["Price"]

# Independent t-test
t_stat, p_value = ttest_ind(more_than_two, two_or_less, equal_var=False)

print("Mean price (>2 bathrooms):", more_than_two.mean())
print("Mean price (≤2 bathrooms):", two_or_less.mean())
print("t-statistic:", t_stat)
print("p-value:", p_value)

Mean price (>2 bathrooms): 1882824.2
Mean price (≤2 bathrooms): 1007347.9350638977
t-statistic: 28.800231483982945
p-value: 7.380428356690378e-137


3. p-value Interpretation:

The p-value = 7.38 × 10⁻¹³⁷ is extremely small and much less than 0.05.
7.38 × 10⁻¹³⁷<0.05

Decision:
Reject the null hypothesis (H₀).

Interpretation:
There is very strong statistical evidence that properties with more than 2 bathrooms sell at higher average prices compared to properties with two or fewer bathrooms.

The large t-statistic (28.8) also indicates a strong difference between the two groups.

4. Recommendation to Policymakers
The results suggest that additional bathrooms significantly increase property value.

Implications for policymakers and developers:
1. Housing developments should consider designing homes with more than two bathrooms to increase property value.
2. Policies encouraging larger or higher-quality housing infrastructure may positively influence property market value.
3. Developers may prioritize multi-bathroom housing projects, especially in high-demand urban areas.
Providing more bathrooms appears to be a strong market feature that buyers value, leading to higher selling prices.

Conclusion:
The independent two-sample t-test shows a statistically significant difference in property prices between homes with more than two bathrooms and those with two or fewer bathrooms.
Since the p-value is far below the significance level of 0.05, we reject the null hypothesis and conclude that properties with more than two bathrooms command a premium price.
    The analysis clearly indicates that bathroom count is an important factor influencing property value, and homes with more than two bathrooms sell at significantly higher prices.